test/unit/string.c

```diff
@@ -4,6 +4,11 @@
 #define UNIT_TAP_COMPATIBLE 1
 #include "unit.h"
 
+static const char *const lower_case = "str";
+static char test_strings[][4] = {
```

Comment:
Nit:

1. Bad name: looks like it's used by all tests while in fact it's needed only for one test.
2. The definition looks strange. Why not use `const char *`?

```C
static const char *test_strtolower_expected = "str";
static const char *test_strtolower_input[] = { ... };
```

Reply:
> while in fact it's needed only for one test

It is used by two tests: `test_strtolower{dup}`.

> Why not use `const char *`?
Because `test_strtolower` modifies `test_strings` (i.e., converts them to lower case in place).

Reply:
I would also prefer moving the data declaration closer to the tests, instead of canonically declaring it at the beginning of the file, before function definitons: what do you think about this?

Reply:
> I would also prefer moving the data declaration closer to the tests, instead of canonically declaring it at the beginning of the file, before function definitons: what do you think about this?

I've no definite opinion regarding this. We do either way in different source files. The test looks good to me now, anyway.